packages/cli/src/build/__tests__/build.ts

```diff
@@ -1135,6 +1139,378 @@ test("correct default export using mjs and dmts proxies", async () => {
+    export {
+      thing
+    } from "./pkg-b.cjs.js";
+
+    ⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯ packages/pkg-b/dist/pkg-b.esm.js ⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯
+    const thing = import('pkg-a').then(x => x.default);
+
+    export { thing };
+
+  `);
+});
+
+test("importing another package via dynamic import and exporting the namespace produces a typescript error because the type cannot be named", async () => {
+  let dir = await testdir({
+    "package.json": JSON.stringify({
+      name: "@mjs-proxy/repo",
+      preconstruct: {
+        packages: ["packages/*"],
+      },
+    }),
+    "packages/pkg-a/package.json": JSON.stringify({
+      name: "pkg-a",
+      main: "dist/pkg-a.cjs.js",
+      module: "dist/pkg-a.esm.js",
+      exports: {
+        ".": {
+          module: "./dist/pkg-a.esm.js",
+          import: "./dist/pkg-a.cjs.mjs",
+          default: "./dist/pkg-a.cjs.js",
+        },
+        "./package.json": "./package.json",
+      },
+      preconstruct: {
+        entrypoints: ["index.ts"],
+        exports: {
+          importDefaultExport: "unwrapped-default",
+        },
+      },
+    }),
+    "packages/pkg-a/src/index.ts": ts`
+      export const thing = "index";
+      export default true;
+    `,
+    "packages/pkg-b/package.json": JSON.stringify({
+      name: "pkg-b",
+      main: "dist/pkg-b.cjs.js",
+      module: "dist/pkg-b.esm.js",
+      exports: {
+        ".": {
+          module: "./dist/pkg-b.esm.js",
+          import: "./dist/pkg-b.cjs.mjs",
+          default: "./dist/pkg-b.cjs.js",
+        },
+        "./package.json": "./package.json",
+      },
+      preconstruct: {
+        entrypoints: ["index.ts"],
+        exports: {
+          importDefaultExport: "unwrapped-default",
+        },
+      },
+      dependencies: {
+        "pkg-a": "^1.0.0",
+      },
+    }),
+
+    "packages/pkg-b/src/index.ts": ts`
+      export const thing = import("pkg-a");
+    `,
+
+    "node_modules/typescript": {
+      kind: "symlink",
+      path: path.join(repoNodeModules, "typescript"),
+    },
+    "tsconfig.json": JSON.stringify({
+      compilerOptions: {
+        module: "NodeNext",
+        moduleResolution: "nodenext",
+        strict: true,
+        declaration: true,
+      },
+    }),
+  });
+  await fs.ensureSymlink(
+    path.join(dir, "packages/pkg-a"),
+    path.join(dir, "node_modules/pkg-a")
+  );
+  await fs.ensureSymlink(
+    path.join(dir, "packages/pkg-b"),
+    path.join(dir, "node_modules/pkg-b")
+  );
+  const error = await build(dir).catch((x) => x);
+  expect(stripAnsi(error + "")).toMatchInlineSnapshot(`
+    "Error: 🎁 Generating TypeScript declarations for packages/pkg-b/src/index.ts failed:
+    🎁 packages/pkg-b/src/index.ts:1:14 - error TS2841: The type of this expression cannot be named without a 'resolution-mode' assertion, which is an unstable feature. Use nightly TypeScript to silence this error. Try updating with 'npm install -D typescript@next'.
+    🎁
+    🎁 1 export const thing = import("pkg-a");
```

Comment:
is this because TS already supported `import` in declaration files, it had its own semantics and stuff so now it can't use that syntax alone to differentiate between loading a thing with import/require semantics? 🤔

Reply:
yes